src/Tools/dotnet-monitor/Egress/Extension/EgressArtifactResult.cs

```diff
@@ -17,8 +17,8 @@ public bool IsValid()
         {
             if (Succeeded)
             {
-                // If Success, we must have an artifact path, and no failure message
-                return !string.IsNullOrEmpty(ArtifactPath) && string.IsNullOrEmpty(FailureMessage);
+                // If Success, we must have no failure message
+                return string.IsNullOrEmpty(FailureMessage);
```

Comment:
When we weren't keeping track of tasks this slipped through the cracks - the artifact path wasn't being set, so the result was never valid. This was causing `_resultCompletionSource` to never set the result in the output parser, which made the `WhenAll` for the validation tasks hang. This is one way of addressing the issue (you could also mandate that validation includes an arbitrary `ArtifactPath`, but I don't think that's a good idea).